src/lightstep_immutable_span_context.cpp

```diff
@@ -0,0 +1,37 @@
+#include "lightstep_immutable_span_context.h"
+
+namespace lightstep {
+//------------------------------------------------------------------------------
+// constructor
+//------------------------------------------------------------------------------
+LightStepImmutableSpanContext::LightStepImmutableSpanContext(
```

Comment:
nit: I would be slightly inclined to remove the LightStep prefix since you're in namespace lightstep.

Reply:
Yeah, it's a little redundant.

I used names like LightStepSpan, LightStepTracer, etc to keep them from clashing with the opentracing names in case you happen to use both namespaces; then chose this name to be consistent. But maybe I should rename them all.

I'll merge in as is for now and maybe make a separate PR to rename everything.